bpf/bpf_lxc.c

```diff
@@ -690,6 +688,22 @@ static inline int __inline__ ipv4_policy(struct __sk_buff *skb, int ifindex, __u
 	if (ret < 0)
 		return ret;
 
+#if defined ENABLE_NAT46 && defined ENABLE_IPV4 && defined CONNTRACK
+	if (skb->cb[CB_NAT46_STATE] == NAT46) {
+		union v6addr dp = LXC_IP;
+
+		ret = ipv4_to_ipv6(skb, ip4, 14, &dp);
+		if (IS_ERR(ret))
+			return ret;
+
+		cilium_trace_capture(skb, DBG_CAPTURE_AFTER_V46, skb->ingress_ifindex);
+
+		ret = ipv6_policy(skb, skb->cb[CB_IFINDEX], skb->cb[CB_SRC_LABEL]);
```

Comment:
`skb->protocol` is changed here, right? It might be better to do another tail call into `__section_tail(CILIUM_MAP_POLICY, LXC_ID)` so we are not inlining the IPv6 policy code segment twice in the program. Did you try that, does the verifier allow for that?